app/Models/SignUp.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Models;
+
+use App\Model;
+
+class SignUp extends Model
+{
+    public function __construct(protected Transaction $transactionModel)
+    {
+        parent::__construct();
+    }
+
+    public function register(array $array): void
+    {
+        try {
+            $this->db->beginTransaction();
+            $this->transactionModel->create($array);
+
+            $this->db->commit();
+        } catch (\Throwable $e) {
+            if ($this->db->inTransaction()) {
+                echo $e->getMessage();
```

Comment:
probably not a good idea to echo exception message from model

Reply:
would be better to either catch exception somewhere in controller & render error page or just log message & return false from here and then return some generic error to the user if register returns false